RELEASE.md

```diff
@@ -9,22 +9,30 @@
 
 ## Migration guide from Kedro 0.18.* to 0.19.*
 
-# Upcoming Release 0.18.11
+# Upcoming Release 0.18.12
 
 ## Major features and improvements
-* Added `databricks-iris` as an official starter.
 
 ## Bug fixes and other changes
-* Reworked micropackaging workflow to use standard Python packaging practices.
-* Make `kedro micropkg package` accept `--verbose`.
 
 ## Documentation changes
-* Added documentation for developing a Kedro project using a Databricks workspace.
 
 ## Breaking changes to the API
 
 ## Upcoming deprecations for Kedro 0.19.0
 
+# Release 0.18.11
+
+## Major features and improvements
+* Added `databricks-iris` as an official starter.
+
+## Bug fixes and other changes
+* Reworked micropackaging workflow to use standard Python packaging practices.
+* Make `kedro micropkg package` accept `--verbose`.
```

Comment:
We'd need to add a line about the renaming of `*DataSet` to `*Dataset`, noting that the old name will emit a `DeprecationWarning` and not be available in 0.19.0 cc @deepyaman

Reply:
As discussed in Slack, #2742 won't be considered for this release and we need to discuss this afterward.

Reply:
Correct - although the `*Dataset` renaming and https://github.com/kedro-org/kedro/pull/2735 should probably go in

Reply:
To update for posterity, we've decided to hold on #2735 for this release; see https://kedro-org.slack.com/archives/C054U0LJLAC/p1688061020135979?thread_ts=1687950117.495209&cid=C054U0LJLAC.